cpu/cortexm_common/thread_arch.c

```diff
@@ -234,6 +237,14 @@ char *thread_arch_stack_init(thread_task_func_t task_func,
     return (char*) stk;
 }
 
+/* This function calculates the ISR_usage */
+int thread_arch_isr_stack_usage(void)
+{
+    uint32_t* sp;
+    asm volatile ("mov %[sp], sp" : [sp] "=r" (sp) : : );
```

Comment:
This will/does not work!

If you call this function from interrupt context, you get the result you are looking for. But if you call it from thread context, you stack pointer is not connected to the ISR stack in any way, so you end up with quite random results. Here is my output when I run this on a `stm32f4discovery` using the default example:
```
2015-11-20 12:20:27,510 - INFO # > ps
2015-11-20 12:20:27,512 - INFO # 	pid | name                 | state    Q | pri | stack ( used) | location | interrupt stack usage
2015-11-20 12:20:27,514 - INFO # 	  1 | idle                 | pending  Q |  15 |   256 (  136) | 0x200001c4 | 896
2015-11-20 12:20:27,523 - INFO # 	  2 | main                 | running  Q |   7 |  1536 (  628) | 0x200002c4 | 896
2015-11-20 12:20:27,525 - INFO # 	    | SUM                  |            |     |  1792 (  764)
```
As you see, the interrupt stack usage is invalid (the stack size is only 512 byte...). Also I think it makes not much sense to print the interrupt stack usage multiple times - remember: we only have one exception (ISR) stack!